File: crossecore-csharp/Java/Java/Ecore/VisibilityKind.cs


namespace Ecore
{

    public class VisibilityKind : EEnumerator
    {
        public const int NONE_VALUE = 1;
        public const int PUBLIC_VALUE = 2;
        public const int PRIVATE_VALUE = 3;
        public const int PROTECTED_VALUE = 4;

        public static VisibilityKind NONE;
        public static VisibilityKind PUBLIC;
        public static VisibilityKind PRIVATE;
        public static VisibilityKind PROTECTED;
        public static VisibilityKind[] VALUES_ARRAY;

        static VisibilityKind()
        {
            VALUES_ARRAY = new VisibilityKind[4];
            NONE = new VisibilityKind(0, "none", "none");
            VALUES_ARRAY[0] = NONE;
        }


        public static VisibilityKind get(string literal)
        {
            for (int i = 0; i < VALUES_ARRAY.Length; ++i)
            {
                VisibilityKind result = VALUES_ARRAY[i];
                if (result.ToString() == literal)
                {
                    return result;
                }
            }
            return null;
        }

        public static VisibilityKind getByName(string name)
        {
            for (int i = 0; i < VALUES_ARRAY.Length; ++i)
            {
                VisibilityKind result = VALUES_ARRAY[i];
                if (result.getName()==name)
                {
                    return result;
                }
            }
            return null;
        }

        public static VisibilityKind get(int value)
        {
            switch (value)
            {

                case NONE_VALUE: return NONE;

                case PUBLIC_VALUE: return PUBLIC;

                case PRIVATE_VALUE: return PRIVATE;

                case PROTECTED_VALUE: return PROTECTED;
            }
            return null;
        }

        private int value;
        private string name;
	    private string literal;

        private VisibilityKind(int value, string name, string literal)
        {
            this.value = value;
            this.name = name;
            this.literal = literal;
        }

        public string getLiteral()
        {
            return literal;
        }

        public string getName()
        {
            return name;
        }

        public int getValue()
        {
            return value;
        }
    }
}

File: crossecore-typescript/java_/VisibilityKind.ts
    export class VisibilityKind
    {
    	
		public static readonly NONE_VALUE:number = 1;
		public static readonly PUBLIC_VALUE:number = 2;
		public static readonly PRIVATE_VALUE:number = 3;
		public static readonly PROTECTED_VALUE:number = 4;
		
		public static NONE:VisibilityKind = new VisibilityKind(1, "none", "none");
		public static PUBLIC:VisibilityKind = new VisibilityKind(2, "public", "public");
		public static PRIVATE:VisibilityKind = new VisibilityKind(3, "private", "private");
		public static PROTECTED:VisibilityKind = new VisibilityKind(4, "protected", "protected");

		private static VALUES_ARRAY:Array<VisibilityKind> = [
			VisibilityKind.NONE, 
			VisibilityKind.PUBLIC, 
			VisibilityKind.PRIVATE, 
			VisibilityKind.PROTECTED
		];

        public static get_string(literal:string):VisibilityKind
        {
            for (let i = 0; i < this.VALUES_ARRAY.length; i++)
            {
                let result = this.VALUES_ARRAY[i];
                if (result.toString() === literal)
                {
                    return result;
                }
            }
            return null;
        }

        public static getByName(name:string):VisibilityKind
        {
	        for (let i = 0; i < this.VALUES_ARRAY.length; i++)
	        {
	            let result = this.VALUES_ARRAY[i];
	            if (result.getName()==name)
	            {
	                return result;
	            }
	        }
	        return null;
        }

        public static get_number(value:number):VisibilityKind
        {
            switch (value)
            {
			case this.NONE_VALUE: return this.NONE;
			case this.PUBLIC_VALUE: return this.PUBLIC;
			case this.PRIVATE_VALUE: return this.PRIVATE;
			case this.PROTECTED_VALUE: return this.PROTECTED;
            }
            return null;
        }

	    private value:number;
	    private name:string;
	    private literal:string;

	    private constructor(value:number, name:string, literal:string)
	    {
	        this.value = value;
	        this.name = name;
	        this.literal = literal;
	    }
	
	    public getLiteral():string
	    {
	        return this.literal;
	    }
	
	    public getName():string
	    {
	        return this.name;
	    }
	
	    public getValue():number
	    {
	        return this.value;
	    }
	    
	    public toString():string
	    {
	        return this.literal;
	    }
    }
